pkg/endpoint/endpoint.go

```diff
@@ -102,6 +102,18 @@ const (
 	// StateRestoring is used to set the endpoint is being restored.
 	StateRestoring = State(models.EndpointStateRestoring)
 
+	// StateLockingDown is used to set the endpoint is attempting
+	// to lockdown.
+	StateLockingDown = State(models.EndpointStateLockingDown)
+
+	// StateSoftLockdown is used to set the endpoint is in a
+	// lockdown mode in which only deny policies are being added.
+	StateSoftLockdown = State(models.EndpointStateAllowLockdown)
+
+	// StateFullLockdown is used to set the endpoint is in a
+	// lockdown mode in which no ingress or egress traffic is allowed.
+	StateFullLockdown = State(models.EndpointStateFullLockdown)
+
```

Comment:
I'd like to see unit tests for the state transitions between these new states and existing states.